haskell-completions.el

```diff
@@ -0,0 +1,158 @@
+;;; haskell-completions.el --- Haskell Completion package
+
+;; FIXME copyright (c) 2015 Haskell mode community. All rights reserved.
+
+;; Authors:
+;; Arthur Fayzrakhmanov <heraldhoi@gmail.com>
+
+;; This file is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation; either version 3, or (at your option)
+;; any later version.
+
+;; This file is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with GNU Emacs; see the file COPYING.  If not, write to
+;; the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+;; Boston, MA 02110-1301, USA.
+
+;;; Commentary:
+
+;; This package provides functionality for all possible auto completions for
+;; Haskell Mode.  FIXME better description.
+
+;;; Code:
+
+(require 'haskell-mode)
+
+(defun haskell-completions-can-grab-prefix ()
+  "Check if the case is appropriate for grabbing completion prefix.
+Returns t if point is either at whitespace character or at line
+end and preceeding character is not a whitespace or new line,
+otherwise returns nil."
+  ;; TODO Discuss this. In some cases it is better to provide completions if
+  ;; point is not at whitespace, for example when point is at closing
+  ;; parenthesis: (ma|) - point is after `a` character.
```

Comment:
In some cases it is better to provide completions if point is not at whitespace, for example when point is at closing parenthesis: `(ma|)` - point is after `a` character. Any thoughts?